deploy/crds/saas.3scale.net_apicasts_crd.yaml

```diff
@@ -0,0 +1,353 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: apicasts.saas.3scale.net
+spec:
+  group: saas.3scale.net
+  names:
+    kind: Apicast
+    listKind: ApicastList
+    plural: apicasts
+    singular: apicast
+  scope: Namespaced
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      type: object
+      description: Apicast is the Schema for Apicast instance
+      properties:
+        apiVersion:
+          description:
+            "APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources"
+          type: string
+        kind:
+          description:
+            "Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds"
+          type: string
+        metadata:
+          type: object
+        spec:
+          type: object
+          description: ApicastSpec defines the desired state of Apicast
+          required:
+            - production
+            - staging
+          properties:
+            production:
+              type: object
+              required:
+                - image
+                - env
+                - marin3r
+                - externalDnsHostname
+              properties:
+                externalDnsHostname:
+                  type: string
+                  description: DNS hostname to manage on AWS Route53 by external-dns
+                image:
+                  type: object
+                  required:
+                    - tag
```

Comment:
On all other controllers, we are adding a default value as image name and tag, trying to use, if possible, a kind of floating tag like master/lates, so there is no need to have it as required value on the CRD, so final user doesn't need to know it and not specify it on a  simple CR.

Why have you configured it as required with no default value?

Reply:
There is no floating tag in the apicast-cloud-hosted repo. In fact is a repo with 2 images (the gateway itself and the mapping-service) so something like `latest` can't be used.

Reply:
@slopezz thoughts?

Reply:
I have checked quay repo, and it seems there are 2 master images on both apicast and mapping service from 3 motnhs ago:

![image](https://user-images.githubusercontent.com/41513123/88062814-97fa7680-cb69-11ea-8e55-7864d887328f.png)

Which is weird, because they are older than a few newer images (2 months ago).

Checking apicast-cloud-hosted  circleci config on GH repo, it seems it was only configured to run with tags, commit from 3 months ago, where master images stopped being created:
https://github.com/3scale/apicast-cloud-hosted/commit/a79612fa6e8aa453d4223012849c6932713e041d#diff-1d37e48f9ceff6d8030570cd36286a61

Maybe it should be added the job to run also with every merge to master branch, so we can have always specific tag images, but also floating master/latest tags.

Maybe this is out of the scope of Apicast controller for saas-operator, we can change it on further versions, and:
- Open issue to create also floating master images on apicast-cloud-hosted repo, and configure now default master images, so tag won't be a required field
- Leave tag as required by the moment, open the issue on  apicast-cloud-hosted repo, and we can update Apicast controller in the future.


What do you think  @roivaz @raelga ?

Reply:
Before I updated the whole apicast-cloud-hosted only master images where created. I changed this to have fixed version images. I really don't see what we gain by having floating tags in the repos an default values in the image.tag field. We are never going to use the default value ...

Reply:
Well, normally on operators you don't need to worry about which app image tags needs to be deployed, all of them comes with default values, normally with possibility for being overriden. So if there is an operator upgrade, it is very possible it will update also app image tag with compatible config... 

However, I agree that on this particular case,  actually `image.tag` is something will need always  to be overriden on all of our saas workloads (field managed by developers), so let's leave it as required without default value, as it is now.